### PR TITLE
Event Handler for AMA channel

### PR DESCRIPTION
**Pull request by kata-gatame**
This PR contains:
- update to the project package:
     - removed `yargs`
     - updated `dev` script
     - updated deps
- update to `Configuration`
     - adds `amaChannel` property
- adds an `Event Handler`
     - monitors a config-specified `amaChannel`
     - reacts to any message with a specified emoji, currently 😄

### CODE REVIEW

File: src/Infrastructure/Client/Rocinante.ts [hunk 1]
```diff
@@ -81,6 +85,7 @@ export default class Rocinante extends Client {
 			await this.roleManager.init();
 			await this.establishedMemberHandler.init();
 			this.commandHandler.init([ APoD, XKCD, Avasarala, Miller ]);
+			await this.eventHandler.init();
```
Comment (johndaniels) [inline src/Infrastructure/Client/Rocinante.ts:88]: We don't need to await on this.

File: src/Infrastructure/Client/Config.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import * as fs from 'fs';
-import * as yargs from 'yargs';
 import * as JSON5 from 'json5';
 import Configuration from '../System/Configuration';
 
 /** @internal */
-const rawConfig: string = fs.readFileSync(yargs.argv.config as string, 'utf8');
+const rawConfig: string = fs.readFileSync('./config.json5' as string, 'utf8');
```
Comment (johndaniels) [inline src/Infrastructure/Client/Config.ts:6]: Did we decide to hardcode this so we could get rid of yargs?

**Review by johndaniels — CHANGES_REQUESTED** [review]

File: src/Infrastructure/Client/Config.ts (hunk 2 above)
Comment (kata-gatame) [reply src/Infrastructure/Client/Config.ts:6]: I was getting errors with the yargs build, also it seemed like hard-coding it wouldn't be an issue.  One less dep.

File: src/Infrastructure/Client/Rocinante.ts (hunk 1 above)
Comment (kata-gatame) [reply src/Infrastructure/Client/Rocinante.ts:88]: Oops, I will update this.

File: src/Infrastructure/Handlers/EventHandler.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Message } from 'discord.js';
+import Rocinante from '../Client/Rocinante';
+
+/**
+ * @category Handler
+ */
+export class EventHandler {
```
[the remark this answers is not in the document]
Comment (kata-gatame) [reply src/Infrastructure/Handlers/EventHandler.ts:7]: Yes, I will update it.

**Review by johndaniels — APPROVED** [review]

**Pull request merged**